src/subdomains/staking/domain/entities/reward-batch.entity.ts

```diff
@@ -37,7 +37,7 @@ export class RewardBatch extends IEntity {
 
     this.rewards = [...(this.rewards ?? []), reward];
 
-    this.outputReferenceAmount = Util.round((this.outputReferenceAmount ?? 0) + reward.outputReferenceAmount, 8);
+    this.outputReferenceAmount = Util.round((this.outputReferenceAmount ?? 0) + reward.outputReferenceAmount, 16);
```

Comment:
This should not be required. `reward.outputReferenceAmount` is rounded by 8 digits and summing up makes it even bigger.